File: Benchmarks/enterprise/python/fastapi/testcode/benchmark_test_22821.py
# SPDX-License-Identifier: Apache-2.0
from fastapi import Request


async def BenchmarkTest22821(request: Request):
    host_value = request.headers.get('host', '')
    data = host_value if host_value else 'default'
    with open('output.csv', 'a') as fh:
        fh.write(str(data) + ',data\n')
    return {"updated": True}

File: Benchmarks/enterprise/java/spring/testcode/BenchmarkTest22821.java
// SPDX-License-Identifier: Apache-2.0
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import org.springframework.web.bind.annotation.*;

@RestController
public class BenchmarkTest22821 {

    @PostMapping(path="/BenchmarkTest22821", consumes="application/xml")
    public void BenchmarkTest22821(@RequestBody String xmlBody, HttpServletRequest request, HttpServletResponse response) throws Exception {
        String xmlValue = xmlBody;
        StringBuilder payload = new StringBuilder();
        payload.append(xmlValue);
        String data = payload.toString();
        response.setContentType("application/json");
        response.getWriter().print("{\"echo\":\"" + data + "\"}");
    }
}
